grails-app/services/au/org/ala/ecodata/ElasticSearchService.groovy

```diff
@@ -1166,6 +1166,18 @@ class ElasticSearchService {
             // todo: Check if BioCollect requires all sites in `sites` property. If no, merge `projectArea` with `sites`.
             projectMap.projectArea = siteService.getSimpleProjectArea(projectMap.projectSiteId)
             projectMap.containsActivity = activityService.searchAndListActivityDomainObjects([projectId: projectMap.projectId], null, null, null, [max: 1, offset: 0])?.totalCount > 0
+
+            List projectActivities = projectActivityService.search([projectId: projectMap?.projectId])
+            if (projectActivities?.size() > 0) {
+                boolean publicParticipation = true
+                projectActivities.each {projectActivity ->
+                    if (projectActivity.publicAccess == false) {
+                        publicParticipation = false
+                    }
+                }
+                projectMap.publicParticipation = publicParticipation
+                projectMap.numberOfRecords = projectActivities.size()
```

Comment:
Use something like `Activity.countByProjectIdAndSatusNotEquals()`. You could add this method to activity service and call that from here.